block_multiblock.php

```diff
@@ -134,6 +137,45 @@ public function add_default_blocks() {
         }
     }
 
+    /**
+     * Clean up incorrect parentcontextid entered by tool_custompages on page duplication.
+     *
+     * @param int $contextid
+     * @return void
+     */
+    private function cleanup($contextid) {
+        global $DB;
```

Comment:
is it possible to add something like the  if ($instance->pagetypepattern == 'admin-tool-custompage') { check in here so that we don't have to check if there a duplicates on every block load? - or maybe even an if tool_custompage plugin exists kind of thing?

Reply:
this is done